pandas/plotting/_core.py

```diff
@@ -2852,22 +2852,46 @@ def pie(self, y=None, **kwds):
 
     def scatter(self, x, y, s=None, c=None, **kwds):
         """
-        Scatter plot
+        A scatter plot with point size `s` and color `c`.
+
+        The coordinates of each point `x,y` are defined by two dataframe
+        columns and filled circles are used to represent each point.
 
         Parameters
         ----------
-        x, y : label or position, optional
-            Coordinates for each point.
+        x : column name or column position
+            Horizontal coordinates of each point.
+        y : column name or column position
+            Vertical coordinates of each point.
         s : scalar or array_like, optional
             Size of each point.
-        c : label or position, optional
+        c : label, column name or column position, optional
```

Comment:
I think valid options are matplotlib's + a column label, which extracts the array and passes it to matplotlib.

from plt.scatter:

```
c : color, sequence, or sequence of color, optional, default: 'b'
    `c` can be a single color format string, or a sequence of color
    specifications of length `N`, or a sequence of `N` numbers to be
    mapped to colors using the `cmap` and `norm` specified via kwargs
    (see below). Note that `c` should not be a single numeric RGB or
    RGBA sequence because that is indistinguishable from an array of
    values to be colormapped.  `c` can be a 2-D array in which the
    rows are RGB or RGBA, however, including the case of a single
    row to specify the same color for all points.
```

You'll to verify that. Check if / when a colorbar is added.